src/client/search-init.ts

```diff
@@ -10,22 +10,21 @@ input.addEventListener("focus", load, {once: true});
 input.addEventListener("keydown", load, {once: true});
 
 // Focus on meta-K and /
-const toggle = document.querySelector("#observablehq-sidebar-toggle")!;
+const toggle = document.querySelector<HTMLInputElement>("#observablehq-sidebar-toggle")!;
 addEventListener("keydown", (event) => {
   if (
     (event.code === "KeyK" && event.metaKey && !event.altKey && !event.ctrlKey) ||
     (event.key === "/" && !event.metaKey && !event.altKey && !event.ctrlKey && event.target === document.body)
   ) {
     // Force the sidebar to be temporarily open while the search input is
-    // focused. (We can’t use :focus-within because the sidebar isn’t focusable
-    // while it is invisible, and we don’t want to keep the sidebar open
-    // persistently after you blur the search input.)
-    toggle.classList.add("observablehq-sidebar-open");
+    // focused. We click on the toggle if necessary to make the input visible so
+    // it be focused. Then the :focus-within selector takes over and keeps it
+    // open until the user blurs the input.
+    const check = getComputedStyle(input).visibility === "hidden";
```

Comment:
Would it be simpler to just look at `toggle.checked` here? As in, force the sidebar open if it’s not already forced open?

Reply:
Yes this works too. I'm also tempted to write this like so:

```
    if (!toggle.checked) toggle.click(), input.focus(), toggle.click();
    else input.focus();
```

it duplicates the main action (focus), but it's much clearer in a sense.